src/script/lua_api/l_env.cpp

```diff
@@ -1376,6 +1376,29 @@ int ModApiEnv::l_get_translated_string(lua_State * L)
 	return 1;
 }
 
+// change_abm(abm_name, parameters)
+int ModApiEnv::l_change_abm(lua_State *L)
+{
+	GET_ENV_PTR;
+
+	std::string abm_name = readParam<std::string>(L, 1);
+
+	ActiveBlockModifier *abm = env->getActiveBlockModifierNoEx(abm_name);
+	if (!abm) {
+		throw LuaError("ABM not found.");
+		return 0;
+	}
+
+	float interval = getfloatfield_default(L, 2, "interval", abm->getTriggerInterval());
```

Comment:
The implementation of `getfloatfield_default` would silently do nothing if this is of a wrong type, wouldn't it? I think it should throw a `LuaError` instead.

Reply:
Yes, it does nothing. It is the same behavior as in `readABMs` functions https://github.com/luanti-org/luanti/blob/4ebaea8b492220b86084cd736d8885d682073c9a/src/script/cpp_api/s_env.cpp#L248

Do we want to throw LuaErrors also from `readABMs` and `readLBMs` functions? If yes, it is a task for different PR I think.

Reply:
Would be good but that's for a different PR for sure.

Reply:
The problem if we use our current `get*field*` methods in new code is that such a subsequent PR will have trouble because there may be legacy code out there that relies on the current behavior.

I think the best way forward is:

1. Add strict, templated variant of `get*field*`. I'd call it `getField<T>`; `T` could be `std::optional<T>` if `nil` is to be allowed.
2. Only use this new variant in new code (such as this PR). This is fine because there is no old mod code that relies on the old behavior.
3. Deprecate the current `get*field*` behavior properly.
4. Eventually, replace all `get*field*` calls with `getField<T>` calls.

TL;DR: Fresh features should not inherit the weird legacy behavior unnecessarily.

Reply:
@appgurueu It sounds like a complex task to add new API functions and use them in many files where Lua functions are implemented and some Lua-type checks should be done.
I do not think that it is a good idea to do it in this PR.
And implementing it will cause many legacy code problems.

So, should I or would you create PR for it and make this PR waiting for dependency?